### PR TITLE
add a browser testing checklist

### PR DESCRIPTION
**Pull request by ascott1**
This adds the browser testing checklist from the Capital Framework docs. Once we all agree on this document, I'll update the docs to point to it so that we have a single source of truth.

### CODE REVIEW

File: browser-checklist.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Browsers
+
+- [ ] Chrome
+- [ ] Safari
+- [ ] FF
+- [ ] IE10
```
Comment (anselmbradford) [inline browser-checklist.md]: I assume un-versioned entries mean latest and greatest. Do you want IE11 included?

File: browser-checklist.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] IE9
+- [ ] IE8
+- [ ] IE7
+- [ ] Opera
+- [ ] iOS8
+- [ ] Android 5
```
Comment (anselmbradford) [inline browser-checklist.md]: Android browser or Chrome for Android?

Comment (ascott1) [inline browser-checklist.md]: It's probably worth testing both? Though to be honest, I just use the Chrome device emulation for this... is anyone testing on a non-Chrome Android browser? If so, how?

File: browser-checklist.md (hunk 1 above)
Comment (ascott1) [inline browser-checklist.md]: IE11 should be added.

File: browser-checklist.md (hunk 2 above)
Comment (anselmbradford) [inline browser-checklist.md]: SauceLabs provides Android 4 through 5.1, which I presume is the Android browser, which is based on Blink—the same engine Chrome for Android uses. I'm not sure what the roadmap is for having both on Android. I'm guessing maybe Chrome will be the default at some point? http://caniuse.com/ lists both, but that may be to cover legacy support, as opposed to feature-support differences. I handled a Samsung Galaxy Tab recently that came with both installed out of the box.

IIRC I was not able to get SauceConnect to work with SauceLab's Android VM to test local changes.

The other option is to install the [Android emulator](http://developer.android.com/tools/help/emulator.html) locally or in a VM, or have an actual Android device available on a local network and test via Ghostlab. 

Chrome device emulation is probably fine, particularly if Chrome for Android is the way things are going.

File: browser-checklist.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+- [ ] Keyboard friendly
+- [ ] Screen reader friendly
+
+## Other
+
+- [ ] Is useable without CSS
+- [ ] Is useable without JS
+- [ ] Flexible from small to large screens
+- [ ] No linting errors or warnings
```
Comment (anselmbradford) [inline browser-checklist.md]: Warning are okay I think, that's the point of having errors _and_ warning level linting, anything that absolutely should not pass should be at an error level.

**Comment by jimmynotjim** [comment]
This is a great baseline. Should we have a guide on how to test for each section, especially for people who have never done testing (I've only done a bit of accessibility testing).

**Comment by ascott1** [comment]
> Should we have a guide on how to test for each section, especially for people who have never done testing (I've only done a bit of accessibility testing).

Yes, definitely. This comes straight from the [CF checklist](http://cfpb.github.io/capital-framework/contributing/testing-snippet.html), which was designed to be copy and pasted into issues/pull requests. This should probably be expanded beyond that though.

File: browser-checklist.md (hunk 3 above)
Comment (ascott1) [inline browser-checklist.md]: Agreed.

**Comment by anselmbradford** [comment]
@ascott1 if you haven't seen it, [GHE]/flapjack/Getting-started-with-Flapjack/wiki#front-end has a lot of different types of testing resources.

File: browser-checklist.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] JavaScript tests are passing
```
Comment (Scotchester) [inline browser-checklist.md]: JS tests and CSS/JS linting are not part of "browser testing" per se. I think they're both better covered under something like #38.

Comment (ascott1) [inline browser-checklist.md]: I agree

**Comment by ascott1** [comment]
> @ascott1 if you haven't seen it, [GHE]/flapjack/Getting-started-with-Flapjack/wiki#front-end has a lot of different types of testing resources.

Whoa, that document is AMAZING. Should we close this in favor of porting that over? Is there value in having both documents?

**Comment by ascott1** [comment]
I made the updates suggested, but let's answer my question above before merging.

**Comment by anselmbradford** [comment]
> Whoa, that document is AMAZING. Should we close this in favor of porting that over? Is there value in having both documents?

I don't think so. At least not beyond flapjack having flapjack-specific content. I think it would be good to bring the general stuff out into the open for sure. Doesn't necessary have to happen now, but up to whatever works for you @ascott1

**Comment by ascott1** [comment]
I say we gradually add that stuff here... at least the stuff that we make use of.

**Comment by anselmbradford** [comment]
Yeah, it's a living doc :)

**Comment by Scotchester** [comment]
Word. :shipit:

**Comment by ascott1** [comment]
Nice. Push the button @Scotchester or @anselmbradford

**Pull request merged**